### PR TITLE
ci: fix subprocess wrapper

### PR DESCRIPTION
**Pull request by gnufede**
This change fixes the `pytest.mark.subprocess` decorator

## Checklist
- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by P403n1x87** [comment]
🙌 great! Any chance that we could introduce some meta-testing? Could we introduce some pytest plugin tests to catch these kind of regressions in our test utilities? I'm thinking something inside say `tests/meta` that we can run as a `pre_check`

File: tests/meta/test_conftest.py
Comment (P403n1x87) [inline tests/meta/test_conftest.py:1]: 👍 👍  thanks for these tests!

**Review by P403n1x87 — COMMENTED** [review]
🙌 Awesome. Just need the changes to `gen_circleci_config.py` and the `riotfile.py` to have the new meta-tests run as pre-checks when the `conftest` sources change, otherwise LGTM!

File: scripts/gen_circleci_config.py [hunk 1]
```diff
@@ -79,6 +79,11 @@ def check(name: str, command: str, paths: t.Set[str]) -> None:
         command="riot -v run -s scripts",
         paths={"scripts/*.py"},
     )
+    check(
+        name="Run conftest tests",
+        command="riot -v run -s conftest",
```
Comment (P403n1x87) [inline scripts/gen_circleci_config.py:84]:
```suggestion
        command="riot -v run meta-testing",
```

File: riotfile.py [hunk 2]
```diff
@@ -121,6 +121,14 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
             "scripts/needs_testrun.py "
             "tests/suitespec.py",
         ),
+        Venv(
+            pys=["3"],
+            name="conftest",
```
Comment (P403n1x87) [inline riotfile.py:126]: nit
```suggestion
            name="meta-testing",
```
?

File: scripts/gen_circleci_config.py [hunk 3]
```diff
@@ -79,6 +79,11 @@ def check(name: str, command: str, paths: t.Set[str]) -> None:
...
+        paths={"tests/conftest.py", "tests/meta/*"},
```
Comment (P403n1x87) [inline scripts/gen_circleci_config.py:85]:
```suggestion
        paths={"tests/*conftest.py", "tests/meta/*"},
```
should future-proof this?

**Review by P403n1x87 — APPROVED** [review]
🙌

**Review by majorgreys — APPROVED** [review]

**Review by emmettbutler — DISMISSED** [review]

**Review by P403n1x87 — APPROVED** [review]

File: tests/profiling/test_profiler.py [hunk 4]
```diff
@@ -181,7 +181,7 @@ def test_disable_memory():
 
 
 @pytest.mark.subprocess(
-    env=dict(DD_PROFILING_AGENTLESS="true", DD_API_KEY="foobar"),
+    env=dict(DD_PROFILING_AGENTLESS="true", DD_API_KEY="foobar", DD_SITE="datadoghq.com"),
```
Comment (P403n1x87) [inline tests/profiling/test_profiler.py:184]:
```suggestion
    env=dict(DD_PROFILING_AGENTLESS="true", DD_API_KEY="foobar", DD_SITE=None),
```

**Review by P403n1x87 — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]
cool, testing conftest helpers 👍🏻 I like it.

**Pull request merged**